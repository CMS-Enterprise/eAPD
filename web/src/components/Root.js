import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import App from './App';
import UpgradeBrowser from './UpgradeBrowser';
import { isBrowserOutdated } from '../util/browser';

const Root = ({ history, store }) => {
  
  return (
    <Provider store={store}>
    { isBrowserOutdated() && <UpgradeBrowser /> }
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>
  );
};

Root.propTypes = {
  history: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  store: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default Root;
