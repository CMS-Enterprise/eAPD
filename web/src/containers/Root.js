import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { setLatestActivity } from '../redux/actions/auth';

import App from './App';

const Root = ({ history, store }) => {
  // Create listener for location changing to track activity
  history.listen(() => {
    store.dispatch(setLatestActivity());
  });

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>
  );
};

Root.propTypes = {
  history: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  store: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default Root;
