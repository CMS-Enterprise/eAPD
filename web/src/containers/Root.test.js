import { shallow } from 'enzyme';
import React from 'react';
import configureMockStore from 'redux-mock-store';

import Root from './Root';

const mockStore = configureMockStore([]);

import { mockFlags, resetLDMocks } from 'jest-launchdarkly-mock';

describe('the Root component', () => {
  const props = { history: { listen: jest.fn() }, store: mockStore({}) };
  beforeEach(() => {
    resetLDMocks();
    mockFlags({ adminCheckFlag: true });
  });

  test('renders correctly', () => {
    const component = shallow(<Root {...props} />);
    expect(component).toMatchSnapshot();
  });
});
