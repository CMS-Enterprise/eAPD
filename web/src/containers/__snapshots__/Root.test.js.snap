// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Root component renders correctly 1`] = `
<LDProvider
  clientSideID="632b257a7cb8135b685301f2"
  options={
    Object {
      "baseUrl": "https://clientsdk.launchdarkly.us",
      "eventsUrl": "https://events.launchdarkly.us",
      "streamUrl": "https://clientstream.launchdarkly.us",
    }
  }
  reactOptions={
    Object {
      "sendEventsOnFlagRead": true,
      "useCamelCaseFlagKeys": true,
    }
  }
>
  <Root
    history={
      Object {
        "listen": [MockFunction],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</LDProvider>
`;
